README.md

```diff
@@ -15,19 +17,86 @@
+Visit the full docs [here](https://PrefectHQ.github.io/prefect-email) to see additional examples and the API reference.
+
+`prefect-email` is a collection of prebuilt Prefect integrations that can be used to interact with email services.
 
 ## Getting Started
 
-### Python setup
+### Integrate with Prefect flows
 
-Requires an installation of Python 3.7+.
+`prefect-email` makes sending emails effortless, giving you peace of mind that your emails are being sent as expected.
 
-We recommend using a Python virtual environment manager such as pipenv, conda or virtualenv.
+First, install [prefect-email](#installation) and [save to block](#saving-credentials-to-block) to run the examples below!
 
-These tasks are designed to work with Prefect 2.0. For more information about how to use Prefect, please refer to the [Prefect documentation](https://orion-docs.prefect.io/).
+```python
+from prefect import flow
+from prefect_email import EmailServerCredentials, email_send_message
+
+@flow
+def example_email_send_message_flow(email_addresses: List[str]):
+    email_server_credentials = EmailServerCredentials.load("BLOCK-NAME-PLACEHOLDER")
+    for email_address in email_addresses:
+        subject = email_send_message.with_options(name=f"email {email_address}").submit(
+            email_server_credentials=email_server_credentials,
+            subject="Example Flow Notification using Gmail",
+            msg="This proves email_send_message works!",
+            email_to=email_address,
+        )
+
+example_email_send_message_flow(["EMAIL-ADDRESS-PLACEHOLDER"])
+```
+
+Outputs:
+
+```bash
+16:58:27.646 | INFO    | prefect.engine - Created flow run 'busy-bat' for flow 'example-email-send-message-flow'
+16:58:29.225 | INFO    | Flow run 'busy-bat' - Created task run 'email someone@gmail.com-0' for task 'email someone@gmail.com'
+16:58:29.229 | INFO    | Flow run 'busy-bat' - Submitted task run 'email someone@gmail.com-0' for execution.
+16:58:31.523 | INFO    | Task run 'email someone@gmail.com-0' - Finished in state Completed()
+16:58:31.713 | INFO    | Flow run 'busy-bat' - Finished in state Completed('All states completed.')
+```
+
+Please note, many email services, like Gmail, require an [App Password](https://support.google.com/accounts/answer/185833) to successfully send emails. If you encounter an error similar to `smtplib.SMTPAuthenticationError: (535, b'5.7.8 Username and Password not accepted...`, it's likely you are not using an App Password.
+
+### Capture exceptions and notify by email
+
+Perhaps you want an email notification with the details of the exception when your flow run fails.
+
+`prefect-email` can be wrapped in an `except` statement to do just that!
+
+```python
+from prefect import flow
+from prefect.context import get_run_context
+from prefect_email import EmailServerCredentials, email_send_message
+
+@flow
+def notify_exc_by_email(exc):
+    context = get_run_context()
+    flow_run_name = context.flow_run.name
```

Comment:
Would this get the name of the parent flow, or the flow that the code is currently running in?

Reply:
You're right. This uses the child flow. I'm simply going to remove the flow decorator here.